ql/src/java/org/apache/hadoop/hive/ql/exec/vector/ptf/VectorPTFOperator.java

```diff
@@ -250,13 +253,16 @@ protected VectorizedRowBatch setupOverflowBatch() throws HiveException {
     for (int i = 0; i < outputProjectionColumnMap.length; i++) {
       int outputColumn = outputProjectionColumnMap[i];
       String typeName = outputTypeInfos[i].getTypeName();
-      allocateOverflowBatchColumnVector(overflowBatch, outputColumn, typeName);
+      allocateOverflowBatchColumnVector(overflowBatch, outputColumn, typeName, outputDataTypePhysicalVariations[i]);
     }
 
     // Now, add any scratch columns needed for children operators.
     int outputColumn = initialColumnCount;
+    DataTypePhysicalVariation[] dataTypePhysicalVariations = vOutContext.getScratchDataTypePhysicalVariations();
     for (String typeName : vOutContext.getScratchColumnTypeNames()) {
-      allocateOverflowBatchColumnVector(overflowBatch, outputColumn++, typeName);
+      allocateOverflowBatchColumnVector(overflowBatch, outputColumn, typeName,
+          dataTypePhysicalVariations[outputColumn-initialColumnCount]);
```

Comment:
I would expect the ScratchCol Types not to include the initial outputCols Types. Why do we need the outputColumn-initialColumnCount here?

Reply:
Actually based on the code here, I think we index the scratch column based on the outputColumnNum
https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/exec/vector/VectorizationContext.java#L525
https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/exec/vector/VectorizationContext.java#L800